ee/cron/chunksHostLoads.js

```diff
@@ -0,0 +1,189 @@
+                  Value: config.chunksLoadBalancerDimensionName,
+                },
+                {
+                  Name: 'TargetGroup',
+                  Value: config.chunksTargetGroupDimensionName,
+                },
+              ],
+            },
+            Period: 60,
+            Stat: 'Sum',
+          },
+        },
+      ],
+    })
+    .promise();
+
+  const pageViewsPerSecondMetric = metrics.MetricDataResults.find(
+    (m) => m.Id === PAGE_VIEWS_PER_SECOND
+  );
+  const activeWorkersPerSecondMetric = metrics.MetricDataResults.find(
+    (m) => m.Id === ACTIVE_WORKERS_PER_SECOND
+  );
+  const loadBalancerRequestsPerMinuteMetric = metrics.MetricDataResults.find(
+    (m) => m.Id === LOAD_BALANCER_REQUESTS_PER_MINUTE
+  );
+
+  const maxPageViewsPerSecond = Math.ceil(arrayMax(pageViewsPerSecondMetric.Values));
+  const maxActiveWorkersPerSecond = Math.ceil(arrayMax(activeWorkersPerSecondMetric.Values));
+  const maxLoadBalancerRequestsPerMinute = Math.ceil(
+    arrayMax(loadBalancerRequestsPerMinuteMetric.Values)
+  );
+
+  const desiredInstancesByPageViews = maxPageViewsPerSecond / config.chunksPageViewsCapacityFactor;
+  const desiredInstancesByActiveWorkers =
+    maxActiveWorkersPerSecond / config.chunksActiveWorkersCapacityFactor;
+  const desiredInstancesByLoadBalancerRequests =
+    maxLoadBalancerRequestsPerMinute / config.chunksLoadBalancerRequestsCapacityFactor;
+  const desiredInstances = Math.max(
+    desiredInstancesByPageViews,
+    desiredInstancesByActiveWorkers,
+    desiredInstancesByLoadBalancerRequests,
+    1
+  );
+
+  /** @type {import('aws-sdk').CloudWatch.Dimensions} */
+  const dimensions = [{ Name: 'ServerGroup', Value: config.groupName }];
+
+  await cloudwatch
+    .putMetricData({
+      MetricData: [
+        {
+          MetricName: 'DesiredInstancesByPageViews',
+          Dimensions: dimensions,
+          StorageResolution: 1,
+          Timestamp: new Date(now),
+          Unit: 'Count',
+          Value: desiredInstancesByPageViews,
+        },
+        {
+          MetricName: 'DesiredInstancesByActiveWorkers',
+          Dimensions: dimensions,
+          StorageResolution: 1,
+          Timestamp: new Date(now),
+          Unit: 'Count',
+          Value: desiredInstancesByActiveWorkers,
+        },
+        {
+          MetricName: 'DesiredInstancesByLoadBalancerRequests',
+          Dimensions: dimensions,
+          StorageResolution: 1,
+          Timestamp: new Date(now),
+          Unit: 'Count',
+          Value: desiredInstancesByLoadBalancerRequests,
+        },
+        {
+          MetricName: 'DesiredInstances',
+          Dimensions: dimensions,
+          StorageResolution: 1,
+          Unit: 'Count',
+          Timestamp: new Date(now),
+          Value: desiredInstances,
+        },
+      ],
+      Namespace: 'Chunks',
+    })
+    .promise();
+
+  await setAutoScalingGroupCapacity(config.chunksAutoScalingGroupName, desiredInstances);
```

Comment:
Note that I've added code that will set the desired capacity for the ASG, but it won't be enabled until we actually set `chunksAutoScalingGroupName` in the config. The intent is that we'll let this code run in production for a while and see if it makes accurate predictions about how many chunk servers we'll need under varying amounts of load.

Reply:
I can't wait to see these metrics in prod!